File: medplum-typescript/packages/sdk/src/model/fhir/CarePlan.ts
import { Annotation } from "./Annotation";
import { canonical } from "./canonical";
import { CarePlan_Activity } from "./CarePlan_Activity";
import { code } from "./code";
import { CodeableConcept } from "./CodeableConcept";
import { dateTime } from "./dateTime";
import { Extension } from "./Extension";
import { id } from "./id";
import { Identifier } from "./Identifier";
import { Meta } from "./Meta";
import { Narrative } from "./Narrative";
import { Period } from "./Period";
import { Reference } from "./Reference";
import { ResourceList } from "./ResourceList";
import { uri } from "./uri";

export interface CarePlan {
    resourceType: string;
    id: id | null | undefined;
    meta: Meta | null | undefined;
    implicitRules: uri | null | undefined;
    language: code | null | undefined;
    text: Narrative | null | undefined;
    contained: ResourceList[] | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    identifier: Identifier[] | null | undefined;
    instantiatesCanonical: canonical[] | null | undefined;
    instantiatesUri: uri[] | null | undefined;
    basedOn: Reference[] | null | undefined;
    replaces: Reference[] | null | undefined;
    partOf: Reference[] | null | undefined;
    status: code | null | undefined;
    intent: code | null | undefined;
    category: CodeableConcept[] | null | undefined;
    title: string | null | undefined;
    description: string | null | undefined;
    subject: Reference;
    encounter: Reference | null | undefined;
    period: Period | null | undefined;
    created: dateTime | null | undefined;
    author: Reference | null | undefined;
    contributor: Reference[] | null | undefined;
    careTeam: Reference[] | null | undefined;
    addresses: Reference[] | null | undefined;
    supportingInfo: Reference[] | null | undefined;
    goal: Reference[] | null | undefined;
    activity: CarePlan_Activity[] | null | undefined;
    note: Annotation[] | null | undefined;
}

File: medplum-java/medplum-model/src/main/java/com/fhir/CarePlan.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableCarePlan.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface CarePlan {
  Optional<Meta> meta();

  Optional<uri> implicitRules();

  Optional<Period> period();

  Optional<code> status();

  Optional<List<Identifier>> identifier();

  Optional<List<Annotation>> note();

  Optional<List<Extension>> modifierExtension();

  Optional<List<Reference>> addresses();

  Optional<Narrative> text();

  Optional<List<Reference>> replaces();

  Optional<String> title();

  Reference subject();

  Optional<String> description();

  Optional<Reference> author();

  Optional<List<Extension>> extension();

  Optional<dateTime> created();

  Optional<List<Reference>> basedOn();

  String resourceType();

  Optional<List<Reference>> supportingInfo();

  Optional<code> language();

  Optional<List<uri>> instantiatesUri();

  Optional<List<ResourceList>> contained();

  Optional<List<Reference>> careTeam();

  Optional<List<canonical>> instantiatesCanonical();

  Optional<List<CarePlan_Activity>> activity();

  Optional<Reference> encounter();

  Optional<code> intent();

  Optional<List<Reference>> contributor();

  Optional<List<Reference>> goal();

  Optional<List<Reference>> partOf();

  Optional<List<CodeableConcept>> category();

  Optional<id> id();

  static ImmutableCarePlan.SubjectBuildStage builder() {
    return ImmutableCarePlan.builder();
  }
}
